aptos-move/framework/aptos-stdlib/sources/math_fixed.move

```diff
@@ -137,3 +136,146 @@ module aptos_std::math_fixed {
         assert!((x - y) * mult < x, 0);
     }
 }
+
+module aptos_std::math_fixed64 {
+    use aptos_std::fixed_point64;
+    use aptos_std::fixed_point64::FixedPoint64;
+    use aptos_std::math128;
+
+    /// Abort code on overflow
+    const EOVERFLOW_EXP: u64 = 1;
+
+    /// Natural log 2 in 32 bit fixed point
+    const LN2: u256 = 12786308645202655660;  // ln(2) in fixed 64 representation
+
+    /// Square root of fixed point number
+    public fun sqrt(x: FixedPoint64): FixedPoint64 {
+        let y = fixed_point64::get_raw_value(x);
+        let z = (math128::sqrt(y) << 32 as u256);
+        z = (z + ((y as u256) << 64) / z) >> 1;
+        fixed_point64::create_from_raw_value((z as u128))
+    }
+
+    /// Exponent function with a precission of 9 digits.
+    public fun exp(x: FixedPoint64): FixedPoint64 {
+        let raw_value = (fixed_point64::get_raw_value(x) as u256);
+        fixed_point64::create_from_raw_value((exp_raw(raw_value) as u128))
+    }
+
+    /// Because log2 is negative for values < 1 we instead return log2(x) + 64 which
+    /// is positive for all values of x.
+    public fun log2_plus_64(x: FixedPoint64): FixedPoint64 {
```

Comment:
nit: Why do we need a 64 suffix?

Reply:
this is because it's adding 64 to the result, to guarantee that result is positive (fixed point cannot be negative)